### ISSUE TITLE
Is it possible to do "nested truthy"

### ISSUE THREAD

**Issue by Jaredk3nt** [stoplightio/spectral#1019]
**Chore summary**

Need clarification on if it is possible to do a "nested" truthy or if I need two separate rules. An example of what I am trying to do is:

```
  description: 'responses should include a `Request-Id` header',
  recommended: true,
  type: 'validation',
  severity: 'warn',
  given: '$.responses[*].headers',
  then: {
    field: 'Request-Id',
    function: 'truthy'
  }
```

This works correctly if the "response" has a headers section, however if no headers section is supplied it will *not* throw a warning that there is no `Request-Id`. Is it possible to get a single rule to give me the same warning if there is no header object?

Thanks!

**Tasks**
- [ ] Clarify rule possibilities

**Comment by P0lip** [comment]
Hey.
I'm sorry for the late answer, but yes, it's totally possible.
```js
  description: 'responses should include a `Request-Id` header',
  recommended: true,
  type: 'validation',
  severity: 'warn',
  given: '$.responses[*]',
  then: {
    field: 'headers.Request-Id',
    function: 'truthy'
  }
```
The above rule should do the trick.

**Comment by P0lip** [comment]
I'll close it for now. Feel free to re-open in case the rule still does not what you expect!
